include/seqan3/search/search_result_range.hpp

```diff
@@ -70,31 +67,21 @@ class search_result_range
 
     /*!\brief Constructs a search result range.
      *
-     * \param[in] search_algorithm The search algorithm to invoke on a single query.
-     * \param[in] query_range The range over the queries to be searched.
+     * \param[in] search_executor The search executor that executes the search and provides the search result.
      */
-    search_result_range(search_algorithm_t search_algorithm, query_range_t query_range) :
-        search_algorithm{std::move(search_algorithm)},
-        single_pass_query_range{views::zip(std::views::iota(0), std::move(query_range))}
+    search_result_range(search_executor_t && search_executor) :
+        executor_ptr{std::make_unique<search_executor_t>(std::move(search_executor))}
     {}
     //!\}
 
     /*!\name Iterators
      * \{
      */
 
-    /*!\brief Returns an iterator to the first element of the search result range.
-     * \return An iterator to the first element.
-     *
-     * \details
-     *
-     * The fist invocation of this function will invoke the search for the first query in the underlying range.
-     */
+    //!\brief Returns an iterator to the first element of the search result range.
```

Comment:
I agree that the `\brief` is already quite descriptive, but I think keeping the `\return` would be nice for the documentation. That way the documentation is consistent for all functions in this file (see  `end()` and `next()` for example).